controller/controller_test.go

```diff
@@ -549,7 +549,7 @@ func TestController(t *testing.T) {
 			provisionerName: "foo.bar/baz",
 			provisioner:     newBadTestProvisioner(),
 			expectedClaims: []v1.PersistentVolumeClaim{
-				*newClaim("claim-1", "uid-1-1", "class-1", "foo.bar/baz", "", map[string]string{annBetaStorageProvisioner: "foo.bar/baz", annSelectedNode: "node-wrong"}),
+				*newClaim("claim-1", "uid-1-1", "class-1", "foo.bar/baz", "", map[string]string{annBetaStorageProvisioner: "foo.bar/baz"}),
```

Comment:
I think you need to add a test Node object to the original test cases, and then add new test cases where the Node object is missing.

Reply:
There is a case below: "provision with selected node, but node does not exist"

Do you mean add a check there to make sure the annSelectedNode is removed from the claim?

Reply:
These were the test cases you had to modify and the name no longer makes sense:
"do not remove selectedNode after final error, only the claim"
"do not remove selectedNode if nothing changes"

I think we need to add the Node object to these original test cases so that it's actually testing the path it wants to test. Right now, with this change, we're hitting the Node doesn't exist error first.

Reply:
I was thinking we want to test the annotation is removed after node doesn't exist?

Reply:
What I'm suggesting is to keep the original test cases (where Node should exist but it seems like it's a bug in the test cases), and add a new one for Node not existng.

Reply:
Adjusted